src/track/external-captions-handler.js

```diff
@@ -116,37 +116,76 @@ class ExternalCaptionsHandler extends FakeEventTarget {
     if (!captions) {
       return [];
     }
-    const textTracks = tracks.filter(track => track instanceof TextTrack);
-    let textTracksLength = textTracks.length || 0;
+    const playerTextTracks = tracks.filter(track => track instanceof TextTrack);
+    let textTracksLength = playerTextTracks.length || 0;
     const newTextTracks = [];
     captions.forEach(caption => {
-      const track = new TextTrack({
-        active: !!caption.default,
-        index: textTracksLength++,
-        kind: 'subtitles',
-        label: caption.label,
-        language: caption.language,
-        external: true
-      });
-      const sameLangTrack = textTracks.find(textTrack => caption.language === textTrack.language);
-      this._textTrackModel[caption.language] = {
-        cuesStatus: CuesStatus.NOT_DOWNLOADED,
-        cues: [],
-        url: caption.url,
-        type: caption.type
-      };
-      if (!sameLangTrack) {
-        if (this._player.config.playback.useNativeTextTrack) {
-          this._addNativeTextTrack(track);
-        }
-        newTextTracks.push(track);
+      if (!caption.language) {
+        const error = new Error(Error.Severity.RECOVERABLE, Error.Category.TEXT, Error.Code.UNKNOWN_LANGUAGE, caption);
```

Comment:
Add meaningful error data and not just the caption object. Use a key to describe the object.

Reply:
added a key, i think this with the error code which refers to the configuration docs is clear, no?